engine/service.go

```diff
@@ -562,25 +564,57 @@ func (e *Engine) nuke(ctx context.Context, opts NukeOptions, op errors.Op) error
 
 // resolveServices resolves a list of services from either a list of service names or a playlist name.
 //
-// If both serivceNames and playlistName are provided, an error will be returned. Mixing service names
+// If both serviceNames and playlistName are provided, an error will be returned. Mixing service names
 // is not supported.
 //
 // If neither serviceNames nor playlistName are provided, then behaviour depends on the value of requireOne.
 // In this case, if requireOne is true, an error will be returned since at least one of serviceNames or playlistName
 // was required. Otherwise, both the returned slice and error will be nil, which can be treated as an empty slice
 // of services.
-func (e *Engine) resolveServices(op errors.Op, serviceNames []string, playlistName string, requireOne bool) ([]service.Service, error) {
-	if len(serviceNames) > 0 && playlistName != "" {
+func (e *Engine) resolveServices(op errors.Op, serviceNames []string, playlistName string, serviceTags map[string]string, requireOne bool) ([]service.Service, error) {
+	if len(serviceNames) > 0 && playlistName != "" && len(serviceTags) == 0 {
 		return nil, errors.New(errkind.Invalid, "both service names and playlist name provided", op)
 	}
 	if len(serviceNames) > 0 {
+		for service := range serviceTags {
+			_, err := e.services.Get(service)
+			if err != nil {
+				return nil, errors.Wrap(err, errors.Meta{Reason: "unable to resolve service", Op: op})
+			}
+		}
+
 		services := make([]service.Service, len(serviceNames))
 		for i, name := range serviceNames {
 			s, err := e.services.Get(name)
 			if err != nil {
 				return nil, errors.Wrap(err, errors.Meta{Reason: "unable to resolve service", Op: op})
 			}
+
 			services[i] = s
+			tag := serviceTags[s.Name]
+			if len(tag) > 0 {
+				override := service.ServiceOverride{
+					Mode: "remote",
+					Remote: service.RemoteOverride{
+						Tag: tag,
+					},
+				}
+				override.Remote.Tag = tag
```

Comment:
This seems redundant if you're adding the tag while creating the override

Reply:
good catch, yeah we don't need this assignment